src/libstd/sys/unix/stack_overflow.rs

```diff
@@ -134,17 +134,42 @@ mod imp {
         Handler { _data: MAIN_ALTSTACK };
     }
 
-    pub unsafe fn make_handler() -> Handler {
-        let alt_stack = mmap(ptr::null_mut(),
-                             SIGSTKSZ,
-                             PROT_READ | PROT_WRITE,
-                             MAP_PRIVATE | MAP_ANON,
-                             -1,
-                             0);
-        if alt_stack == MAP_FAILED {
+    #[cfg(any(target_os = "linux",
+              target_os = "macos",
+              target_os = "bitrig",
+              target_os = "netbsd",
+              target_os = "openbsd"))]
+    unsafe fn get_stack() -> *mut libc::c_void {
```

Comment:
I think it would have more sens to make `get_stack()` return a `struct stack_t` as it is the assignment of `stack.ss_sp = alt_stack` which differe on platforms.

The problem is on freebsd/dragonfly, `mmap()` returns a `*mut ::c_void` and `ss_sp` is a `*mut ::c_char`. So a cast is need for freebsd/dragonfly.

It would be less confusing that casting `mmap()`.